src/meteodatalab/metadata.py

```diff
@@ -140,6 +144,26 @@ def compute_origin(ref_grid: Grid, field: xr.DataArray) -> dict[str, float]:
     }
 
 
+def _uses_icon_grid(metadata: Metadata) -> bool:
+    """Determine if the data is on a MeteoSwiss ICON grid.
+
+    Parameters
+    ----------
+    metadata : Metadata
+        GRIB metadata containing the grid definition.
+
+    Returns
+    -------
+    bool
+        True if the data was created by an MCH ICON forecast.
+
+    """
+    return metadata.get("centre", default="") == "lssw" and (
```

Comment:
This indicates that the data was produced by ICON perhaps this should include a check on the grid_type as well. Technically, users can work with regridded data.